internal/fs/vss_windows.go

```diff
@@ -0,0 +1,977 @@
+}
+
+// DeleteSnapshots calls the equivalent VSS api.
+func (vss *IVssBackupComponents) DeleteSnapshots(snapshotID ole.GUID) (int32, ole.GUID, error) {
+	var deletedSnapshots int32 = 0
+	var nondeletedSnapshotID ole.GUID
+	var result uintptr = 0
+
+	if runtime.GOARCH == "386" {
+		id := (*[4]uintptr)(unsafe.Pointer(ole.IID_NULL))
+
+		result, _, _ = syscall.Syscall9(vss.getVTable().deleteSnapshots, 9,
+			uintptr(unsafe.Pointer(vss)), id[0], id[1], id[2], id[3],
+			uintptr(VSS_OBJECT_SNAPSHOT), uintptr(1), uintptr(unsafe.Pointer(&deletedSnapshots)),
+			uintptr(unsafe.Pointer(&nondeletedSnapshotID)),
+		)
+	} else {
+		result, _, _ = syscall.Syscall6(vss.getVTable().deleteSnapshots, 6,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&snapshotID)),
+			uintptr(VSS_OBJECT_SNAPSHOT), uintptr(1), uintptr(unsafe.Pointer(&deletedSnapshots)),
+			uintptr(unsafe.Pointer(&nondeletedSnapshotID)))
+	}
+
+	err := newVssErrorIfResultNotOK("DeleteSnapshots() failed", HRESULT(result))
+	return deletedSnapshots, nondeletedSnapshotID, err
+}
+
+// GetSnapshotProperties calls the equivalent VSS api.
+func (vss *IVssBackupComponents) GetSnapshotProperties(snapshotID ole.GUID,
+	properties *VssSnapshotProperties) error {
+	var result uintptr = 0
+
+	if runtime.GOARCH == "386" {
+		id := (*[4]uintptr)(unsafe.Pointer(ole.IID_NULL))
+
+		result, _, _ = syscall.Syscall6(vss.getVTable().getSnapshotProperties, 6,
+			uintptr(unsafe.Pointer(vss)), id[0], id[1], id[2], id[3],
+			uintptr(unsafe.Pointer(properties)))
+	} else {
+		result, _, _ = syscall.Syscall(vss.getVTable().getSnapshotProperties, 3,
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&snapshotID)),
+			uintptr(unsafe.Pointer(properties)))
+	}
+
+	return newVssErrorIfResultNotOK("GetSnapshotProperties() failed", HRESULT(result))
+}
+
+// vssFreeSnapshotProperties calls the equivalent VSS api.
+func vssFreeSnapshotProperties(properties *VssSnapshotProperties) error {
+	proc, err := findVssProc("VssFreeSnapshotProperties")
+	if err != nil {
+		return err
+	}
+
+	proc.Call(uintptr(unsafe.Pointer(properties)))
+	return nil
+}
+
+// BackupComplete calls the equivalent VSS api.
+func (vss *IVssBackupComponents) BackupComplete() (*IVSSAsync, error) {
+	var oleIUnknown *ole.IUnknown
+	result, _, _ := syscall.Syscall(vss.getVTable().backupComplete, 2, uintptr(unsafe.Pointer(vss)),
+		uintptr(unsafe.Pointer(&oleIUnknown)), 0)
+
+	err := newVssErrorIfResultNotOK("BackupComplete() failed", HRESULT(result))
+	return vss.convertToVSSAsync(oleIUnknown, err)
+}
+
+// VssSnapshotProperties defines the properties of a VSS snapshot as part of the VSS api.
+type VssSnapshotProperties struct {
+	snapshotID           ole.GUID
+	snapshotSetID        ole.GUID
+	snapshotsCount       uint32
+	snapshotDeviceObject *uint16
+	originalVolumeName   *uint16
+	originatingMachine   *uint16
+	serviceMachine       *uint16
+	exposedName          *uint16
+	exposedPath          *uint16
+	providerId           ole.GUID
```

Comment:
golint: struct field providerId should be providerID

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/f70a1dfd172d54843e27deae292f3557d11ade84